src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Properties/InterceptedProjectProperties/BuildPropertyPage/TrimmingEnumProvider.cs

```diff
@@ -0,0 +1,50 @@
+﻿// Licensed to the .NET Foundation under one or more agreements. The .NET Foundation licenses this file to you under the MIT license. See the LICENSE.md file in the project root for more information.
+
+using Microsoft.Build.Framework.XamlTypes;
+using Microsoft.VisualStudio.ProjectSystem.Properties;
+
+namespace Microsoft.VisualStudio.ProjectSystem.VS.Properties;
+
+[ExportDynamicEnumValuesProvider("TrimmingEnumProvider")]
+[AppliesTo(ProjectCapability.DotNet)]
+internal class TrimmingEnumProvider : IDynamicEnumValuesProvider
+{
+
+    [ImportingConstructor]
+    public TrimmingEnumProvider()
+    {
+    }
+
+    public Task<IDynamicEnumValuesGenerator> GetProviderAsync(IList<NameValuePair>? options)
+    {
+        return Task.FromResult<IDynamicEnumValuesGenerator>(new TrimmingEnumGenerator());
+    }
+}
+
+internal class TrimmingEnumGenerator : IDynamicEnumValuesGenerator
```

Comment:
Consider moving this class inside `TrimmingEnumProvider`, as a private nested class. I don't think this needs to be a top level type.